### PR TITLE
Update CloudBigtableIO Java Doc.

### PR DESCRIPTION
**Pull request by kevinsi4508**

### CODE REVIEW

File: bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableIO.java [hunk 1]
```diff
@@ -139,7 +143,7 @@
    */
   static class Source extends BoundedSource<Result> {
     private static final long serialVersionUID = -5580115943635114126L;
-    private static final Logger LOG = LoggerFactory.getLogger(Source.class);
+    private static final Logger SOURCE_LOG = LoggerFactory.getLogger(Source.class);
```
Comment (mgarolera) [inline bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableIO.java]: nit: Why call it SOURCE_? its already in the Source class.

Comment (kevinsi4508) [inline bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableIO.java]: LOG hides the static variable in the parent class.

File: bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableIO.java [hunk 2]
```diff
@@ -165,30 +169,34 @@
...
             "[startKey, stopKey]", new String(startKey), new String(stopKey));
-          Preconditions.checkState(size > 0);
+          Preconditions.checkState(estimatedSize > 0);
         }
         this.startKey = startKey;
         this.stopKey = stopKey;
-        this.estimatedSize = size;
-        LOG.debug("Creating split: {}.", this);
+        this.estimatedSize = estimatedSize;
+        SOURCE_LOG.debug("Source with split: {}.", this);
```
Comment (mgarolera) [inline bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableIO.java]: What does {} represent? should it be {startKey, stopKey}?

Comment (kevinsi4508) [inline bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableIO.java]: it is a placeholder for "this".

File: bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableIO.java [hunk 3]
```diff
@@ -223,8 +233,10 @@ public boolean producesSortedKeys(PipelineOptions options) throws Exception {
       /**
        * <p>Splits the bundle based on the assumption that the data is distributed evenly between
```
Comment (mgarolera) [inline bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableIO.java]: Should the first line be without <p>? We are inconsistent about that.

File: bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableIO.java [hunk 4]
```diff
@@ -437,13 +426,13 @@ public boolean start() throws IOException {
...
-      LOG.info("Creating {} splits.", splits.size());
-      LOG.debug("Created splits {}.", splits);
+      SOURCE_LOG.info("Creating {} splits.", splits.size());
+      SOURCE_LOG.debug("Created splits {}.", splits);
       return splits;
     }
 
     /**
-     * Is the range of the region within the range of the scan?
+     * Checks if the range of the region is within the range of the scan.
```
Comment (mgarolera) [inline bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableIO.java]: nit: Maybe mention that it is entirely contained?

File: bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableIO.java [hunk 5]
```diff
@@ -453,7 +442,7 @@ private static boolean isWithinRange(byte[] scanStartKey, byte[] scanEndKey,
     }
 
     /**
-     * Perform a call to the {@link BigtableDataClient#sampleRowKeys(SampleRowKeysRequest)} that
+     * Performs a call to the {@link BigtableDataClient#sampleRowKeys(SampleRowKeysRequest)} that
```
Comment (mgarolera) [inline bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableIO.java]: mention that the result is cached?

File: bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableIO.java (hunk 3 above)
Comment (kevinsi4508) [inline bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableIO.java]: yeah, not very consistent. updated.

File: bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableIO.java (hunk 4 above)
Comment (kevinsi4508) [inline bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableIO.java]: I don't think it is entirely contains. It is within the scan range. @sduskis , can you confirm?

File: bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableIO.java (hunk 5 above)
Comment (kevinsi4508) [inline bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableIO.java]: done.

File: bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableIO.java (hunk 4 above)
Comment (sduskis) [inline bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableIO.java]: Correct.  The ranges do not have to be entirely overlapping.

**Comment by kevinsi4508** [comment]
Marcal is off to his vacation

File: bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableIO.java (hunk 2 above)
Comment (sduskis) [inline bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableIO.java]: and "this" translates to "this.toString()"

**Comment by sduskis** [comment]
LGTM

**Comment by kevinsi4508** [comment]
Jeff, please also take a look.

**Comment by hegemonic** [comment]
LGTM

**Pull request merged**